pkg/apis/application/v1alpha1/types.go

```diff
@@ -183,6 +183,18 @@ type ApplicationSource struct {
 // ApplicationSources contains list of required information about the sources of an application
 type ApplicationSources []ApplicationSource
 
+func (s ApplicationSources) Equals(other ApplicationSources) bool {
+	if len(s) != len(other) {
+		return false
+	}
+	for i := range s {
+		if !s[i].Equals(other[i]) {
```

Comment:
If the 2 `ApplicationSources` being compared have the same elements but in a different order, is this considered to be equal or not? If it does, the code needs to be adapted to address this case.

Reply:
It's not equal, because that changes the "winner" when there are duplicate resources - the later source always wins.